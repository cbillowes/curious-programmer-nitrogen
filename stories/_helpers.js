import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page