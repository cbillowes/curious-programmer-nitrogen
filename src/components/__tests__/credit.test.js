import React from 'react'
import Credit from '../credit'
import { getSnapshot, getElement, mustNotContainValue, validateValues } from './_helpers'

describe(`Credit`, () => {
  it(`should be empty`, () => {
    const tree = getSnapshot(
      <Credit />
    )
    expect(tree).toMatchSnapshot()
  })

  it(`should only render its body`, () => {
    const tree = getSnapshot(
      <Credit>
        <p>Only this will be rendered.</p>
      </Credit>
    )
    expect(tree).toMatchSnapshot()
  })

  it(`should use the title for its empty body`, () => {
    const tree = getSnapshot(
      <Credit
        title="Use me instead.">
      </Credit>
    )
    expect(tree).toMatchSnapshot()
  })

  it(`should render an external credit`, () => {
    const tree = getSnapshot(
      <Credit
        to="https://curiousprogrammer.dev"
        title="Curious Programmer"
      >
        Curious Programmer
      </Credit>
    )
    expect(tree).toMatchSnapshot()
  })

  it(`should render an internal credit`, () => {
    const tree = getSnapshot(
      <Credit
        to="/blog"
        title="Curious Programmer"
      >
        Curious Programmer
      </Credit>
    )
    expect(tree).toMatchSnapshot()
  })

  describe(`should not`, () => {
    it(`generate an anchor when a link is not provided`, () => {
      const credit = (
        <Credit
          title="hello world."
        >
        </Credit>
      )
      const match = mustNotContainValue(credit, `href="`)
      expect(match).toBe(true)
    })

    it(`link out of the website for internal links`, () => {
      const credit = (
        <Credit
          title="internal"
          to="/blog"
        >
          Internal link
        </Credit>
      )
      const matches = validateValues(credit,
        [
          { contains: false, value: `nofollow noopener noreferrer"` },
          { contains: false, value: `target="_blank"` }
        ])
      expect(matches).toBe(true)
    })

    it(`stay within the website for external links`, () => {
      const credit = (
        <Credit
          title="external"
          to="https://external.link"
        >
          External link
        </Credit>
      )
      const matches = validateValues(credit,
        [
          { contains: true, value: `nofollow noopener noreferrer"` },
          { contains: true, value: `target="_blank"` }
        ])
      expect(matches).toBe(true)
    })
  })
})