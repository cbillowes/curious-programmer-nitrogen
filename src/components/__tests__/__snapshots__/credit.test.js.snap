// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credit should be empty 1`] = `<div />`;

exports[`Credit should only render its body 1`] = `
<div>
  <p>
    Only this will be rendered.
  </p>
</div>
`;

exports[`Credit should render an external credit 1`] = `
<div>
   
  <a
    href="https://curiousprogrammer.dev"
    rel="nofollow noopener noreferrer"
    target="_blank"
    title="Curious Programmer"
  >
    curious programmer
  </a>
   
  Curious Programmer
</div>
`;

exports[`Credit should render an internal credit 1`] = `
<div>
   
  <a
    href="/blog"
    onClick={[Function]}
    onMouseEnter={[Function]}
    title="Curious Programmer"
  >
    curious programmer
  </a>
   
  Curious Programmer
</div>
`;

exports[`Credit should use the title for its empty body 1`] = `
<div>
  Use me instead.
</div>
`;
